nsxt/metadata/metadata.go

```diff
@@ -322,6 +322,10 @@ func SchemaToStruct(elem reflect.Value, d *schema.ResourceData, metadata map[str
 			} else {
 				value = d.Get(key).(bool)
 			}
+			if item.Metadata.OmitIfEmpty && !value {
```

Comment:
wouldn't this be true if the value of bool is specified as `false`?

Reply:
When bool is unspecified it defaults to `false`, it's also evident [here](https://github.com/vmware/terraform-provider-nsxt/blob/master/nsxt/metadata/metadata_test.go#L630) in the tests.
Nonetheless I've revised this PR to use `GetOk()` and handle cases where an attribute is omitted within a sub-object - before it could crash the provider IMO.